src/wallet/connect.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  AccountChangeEventHandler,
+  AddDeclareTransactionParameters,
+  AddDeployAccountTransactionParameters,
+  AddInvokeTransactionParameters,
+  AddStarknetChainParameters,
+  NetworkChangeEventHandler,
+  StarknetChainId,
+  StarknetWindowObject,
+  TypedData,
+  WatchAssetParameters,
+} from './getst/main';
+
+/**
+ * Request Permission for wallet account, return addresses that's allowed by user
+ * @param silentMode false: request user interaction allowance. true: return only pre-allowed
+ * @returns allowed accounts addresses
+ */
+export function requestAccounts(swo: StarknetWindowObject, silentMode = false) {
+  return swo.request({
+    type: 'wallet_requestAccounts',
+    params: {
+      silentMode,
+    },
+  });
+}
+
+/**
+ * Request Permission for wallet account
+ * @returns allowed accounts addresses
+ */
+export function getPermissions(swo: StarknetWindowObject) {
+  return swo.request({ type: 'wallet_getPermissions' });
+}
+
+/**
+ * Request adding ERC20 Token to Wallet List
+ * @param asset WatchAssetParameters
+ * @returns boolean
+ */
+export function watchAsset(swo: StarknetWindowObject, asset: WatchAssetParameters) {
+  return swo.request({
+    type: 'wallet_watchAsset',
+    params: asset,
+  });
+}
+
+/**
+ * Request adding custom Starknet chain
+ * @param chain AddStarknetChainParameters
+ * @returns boolean
+ */
+export function addStarknetChain(swo: StarknetWindowObject, chain: AddStarknetChainParameters) {
+  // Can this set custom RPC endpoint ?
```

Comment:
also maybe TODO: here if still relevant

Reply:
Yeah this one didn't work on the custom wallet build